pkg/reconciler/instances/istio/reset/data/gatherer.go

```diff
@@ -56,12 +56,19 @@ func (i *DefaultGatherer) GetPodsWithDifferentImage(inputPodsList v1.PodList, im
 	outputPodsList.Items = []v1.Pod{}
 
 	for _, pod := range inputPodsList.Items {
+		isIstioPod := false
+		if _, ok := pod.Annotations["sidecar.istio.io/status"]; ok {
+			isIstioPod = true
+		} else {
+			continue
+		}
+		if !isPodReady(pod) {
```

Comment:
Is pod readiness crucial in this scenario? Even if pod is not ready shouldn't it be included as long as it contains different image?